drivers/flash/soc_flash_nrf.c

```diff
@@ -153,12 +157,35 @@ static int flash_nrf_read(const struct device *dev, off_t addr,
 		return 0;
 	}
 
-#if CONFIG_ARM_NONSECURE_FIRMWARE && CONFIG_SPM && USE_PARTITION_MANAGER \
-	&& CONFIG_SPM_SECURE_SERVICES
-	if (addr < PM_APP_ADDRESS) {
+	// If the currently executing image is non-secure and the address
+	// is not in the non-secure image we will not be able to read it
+	// directly and must instead invoke a secure service to read out
+	// the flash.
+#ifdef CONFIG_ARM_NONSECURE_FIRMWARE
+	// We assume that image_0_nonsecure is the only non-secure
+	// partition. This will need to be updated when it is also
+	// possible to execute from slot 1. Partitions that are assumed to
+	// be secure are: mcuboot, image_0, image_scratch, and storage.
+	if (!is_within_bounds(addr, len, FLASH_AREA_OFFSET(image_0_nonsecure),
+			      FLASH_AREA_SIZE(image_0_nonsecure))) {
```

Comment:
I think maybe we can try to map these in FLASH_LAYOUT with PM to what was used before.

Reply:
@sigvartmh Could you include the tfm_read_ranges.h header and check against it instead? That should work for upstream I think.

Reply:
Seems we have the image_0_nonsecure already available downstream from https://github.com/nrfconnect/sdk-nrf/blob/59cd80ab8e3d7422fe2624104258cdf0bd2b26e9/include/flash_map_pm.h#L16

I only found `tfm_read_ranges.h` in the nrf repository.

Reply:
seems the defines are not always available (aka if you are not using mcuboot) in downstream. Added some more checks for it.